tests/socket_test.c

```diff
@@ -432,6 +432,56 @@ static int s_test_tcp_socket_communication(struct aws_allocator *allocator, void
 
 AWS_TEST_CASE(tcp_socket_communication, s_test_tcp_socket_communication)
 
+static int s_test_socket_with_bind_to_interface(struct aws_allocator *allocator, void *ctx) {
+    (void)ctx;
+#if !defined(AWS_OS_APPLE) && !defined(AWS_OS_LINUX)
```

Comment:
since the ifdefs might get crazy, maybe just go through with the test and skip if it fails with PLATFORM_NOT_SUPPORTED

Reply:
> since the ifdefs might get crazy
We can discuss the `ifdefs` in the office tomorrow.